pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go

```diff
@@ -33,30 +33,25 @@ import (
 	v1 "kubevirt.io/api/core/v1"
 
 	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/pkg/pointer"
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks/mutating-webhook/mutators"
 )
 
 var _ = Describe("VirtualMachineInstanceMigration Mutator", func() {
 	It("Should mutate the VirtualMachineInstanceMigration object", func() {
 		migration := newMigration()
 
-		ar, err := newAdmissionReview(migration)
+		admissionReview, err := newAdmissionReview(migration)
 		Expect(err).ToNot(HaveOccurred())
 
 		mutator := &mutators.MigrationCreateMutator{}
-		resp := mutator.Mutate(ar)
-		Expect(resp.Allowed).To(BeTrue())
 
-		migrationMeta := k8smetav1.ObjectMeta{}
-		patchOps := []patch.PatchOperation{
-			{Value: &migrationMeta},
-		}
+		expectedJSONPatch, err := expectedJSONPatch(
+			expectedMigrationObjectMeta(migration.ObjectMeta, migration.Spec.VMIName),
+		)
+		Expect(err).NotTo(HaveOccurred())
 
-		err = json.Unmarshal(resp.Patch, &patchOps)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(patchOps).NotTo(BeEmpty())
-
-		Expect(migrationMeta).To(Equal(expectedMigrationObjectMeta(migration.ObjectMeta, migration.Spec.VMIName)))
+		Expect(mutator.Mutate(admissionReview)).To(Equal(newAllowedAdmissionResponse(expectedJSONPatch)))
```

Comment:
IMO this does not help with the test readability, it makes it harder to understand what is tested.
Seeing directly what data is expected is better than using a helper that hides it.

But I am not too strong opinionated about it.

Reply:
The Admission response is pretty long, contains a JSON string for the patch and is dependent on the VMI's metadata field.

Reply:
The AdmissionReview object is also pretty long:
```json
{
  "kind": "AdmissionReview",
  "apiVersion": "admission.k8s.io/v1",
  "request": {
    "uid": "356c91db-e94c-4c65-bef0-eee99eedf53e",
    "kind": {
      "group": "kubevirt.io",
      "version": "v1",
      "kind": "VirtualMachineInstanceMigration"
    },
    "resource": {
      "group": "kubevirt.io",
      "version": "v1",
      "resource": "virtualmachineinstancemigrations"
    },
    "requestKind": {
      "group": "kubevirt.io",
      "version": "v1",
      "kind": "VirtualMachineInstanceMigration"
    },
    "requestResource": {
      "group": "kubevirt.io",
      "version": "v1",
      "resource": "virtualmachineinstancemigrations"
    },
    "namespace": "default",
    "operation": "CREATE",
    "userInfo": {
      "username": "system:serviceaccount:kubevirt:kubevirt-apiserver",
      "uid": "4f2f8527-d1f7-4052-9b12-d35d3da321c8",
      "groups": [
        "system:serviceaccounts",
        "system:serviceaccounts:kubevirt",
        "system:authenticated"
      ],
      "extra": {
        "authentication.kubernetes.io/pod-name": [
          "virt-api-7b6f6449f6-kqsvq"
        ],
        "authentication.kubernetes.io/pod-uid": [
          "e4cd6d90-f0ae-451e-8598-e88e6aa019ba"
        ]
      }
    },
    "object": {
      "apiVersion": "kubevirt.io/v1",
      "kind": "VirtualMachineInstanceMigration",
      "metadata": {
        "generateName": "kubevirt-migrate-vm-",
        "managedFields": [
          {
            "apiVersion": "kubevirt.io/v1",
            "fieldsType": "FieldsV1",
            "fieldsV1": {
              "f:metadata": {
                "f:generateName": {}
              },
              "f:spec": {
                ".": {},
                "f:vmiName": {}
              }
            },
            "manager": "virt-api",
            "operation": "Update",
            "time": "2024-03-21T08:38:11Z"
          }
        ],
        "namespace": "default"
      },
      "spec": {
        "vmiName": "vm-cirros"
      },
      "status": {}
    },
    "oldObject": null,
    "dryRun": false,
    "options": {
      "kind": "CreateOptions",
      "apiVersion": "meta.k8s.io/v1"
    }
  }
}
```

Reply:
@orelmisan Hi, if the assertion will failed due to objects being unequal, you will have to manually compare the objects, I think its better to assert on each field individually. With that you will know exactly where is the unexpected data.

Reply:
In case the objects are unequal, Gomage shows the diff between the objects.
Testing equality between the objects makes sure that if a field is added / removed / changed - the test will notice.

Reply:
@orelmisan Does it? I remember that it prints you the current and the expected, w/o details. Maybe I am wrong. Will check it

Reply:
```
• [FAILED] [0.001 seconds]
VirtualMachineInstanceMigration Mutator [It] Should mutate the VirtualMachineInstanceMigration object
pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go:41

  [FAILED] Expected
      <*v1.AdmissionResponse | 0xc006786720>: {
          UID: "",
          Allowed: true,
          Result: nil,
          Patch: "[{\"op\":\"replace\",\"path\":\"/metadata\",\"value\":{\"creationTimestamp\":null,\"labels\":{\"kubevirt.io/vmi-name\":\"testVmi\",\"test\":\"test\"},\"finalizers\":[\"kubevirt.io/migrationJobFinalize\"]}}]",
          PatchType: "JSONPatch",
          AuditAnnotations: nil,
          Warnings: nil,
      }
  to equal
      <*v1.AdmissionResponse | 0xc006786780>: {
          UID: "",
          Allowed: false,
          Result: nil,
          Patch: "[{\"op\":\"replace\",\"path\":\"/metadata\",\"value\":{\"creationTimestamp\":null,\"labels\":{\"kubevirt.io/vmi-name\":\"testVmi\",\"test\":\"test\"},\"finalizers\":[\"kubevirt.io/migrationJobFinalize\"]}}]",
          PatchType: "JSONPatch",
          AuditAnnotations: nil,
          Warnings: nil,
      }
  In [It] at: pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go:54 @ 03/26/24 12:46:50.024
```
Using diff:
```
1c1
<       <*v1.AdmissionResponse | 0xc006786720>: {
---
>       <*v1.AdmissionResponse | 0xc006786780>: {
3c3
<           Allowed: true,
---
>           Allowed: false,
```

Reply:
> IMO this does not help with the test readability, it makes it harder to understand what is tested. Seeing directly what data is expected is better than using a helper that hides it.
> 
> But I am not too strong opinionated about it.

Done.